src/core/toga/widgets/table.py

```diff
@@ -53,7 +53,10 @@ def __init__(self, headings, id=None, style=None, data=None, accessors=None,
         self._on_select = None
         self._selection = None
         self._data = None
-        self._missing_value = missing_value
+        if not missing_value:
+            print("WARNING: Using empty string for missing value in data. "
+                  "Define a 'missing_value' on the table to silence this message")
+        self._missing_value = missing_value or ''
```

Comment:
This will only be needed *if* there is actually missing data in the table source. Having the warning raised even when missing_value isn't needed isn't a great UX. The previous approach would only raise a warning if it was actually a problem.